addons/event/tests/test_event_internals.py

```diff
@@ -58,6 +58,27 @@ def test_event_date_computation(self):
         })
         self.assertTrue(event.is_one_day)
 
+        # Checks case when mocked today changes date before event, when event.date_tz considered
+        self.mock_datetime.now.return_value = datetime(2020, 6, 20, 20, 0, 0)
+        event.write({
+            'date_begin': datetime(2020, 6, 27, 1, 0, 0),
+            'date_end': datetime(2020, 7, 8, 2, 0, 0),
+            'date_tz': 'America/Los_Angeles'
+        })
+        self.assertEqual(registration.get_date_range_str(), u'in 6 days')
+
+        # Checks case when event changes date before mocked today, when event.date_tz considered
+        self.mock_datetime.now.return_value = datetime(2020, 6, 20, 13, 0, 0)
+        event.write({
+            'date_begin': datetime(2020, 6, 25, 20, 0, 0),
+            'date_end': datetime(2020, 7, 8, 2, 0, 0),
+            'date_tz': 'Australia/Sydney'
+        })
+        self.assertEqual(registration.get_date_range_str(), u'in 6 days')
+
+        # Resets mocked 'Today' value back to original
+        self.mock_datetime.now.return_value = datetime(2020, 1, 31, 10, 0, 0)
```

Comment:
As the value should be reset at the beginning of each tests, do we really need to do this ?

Reply:
I thought this would be the case as well, but it doesn't seem to reset as we'd expect. So I had to manually reset it to avoid subsequent tests from failing.

![image](https://github.com/odoo/odoo/assets/121266888/66dd4a1f-4bff-46a7-ab95-691cbad5b87b)
![image](https://github.com/odoo/odoo/assets/121266888/cc732187-57d0-4e7e-8a02-964ed9c673e3)

Reply:
Ok thanks for having look, a bit weird so I note this for me to investigate later when I have some time :+1:

Reply:
It's a class setup so it seems to make sense
Seems like a weird approach anyway, I don't think we use this in master anymore so it will probably break along the way ^^

Reply:
Yeop, immediately in 16 lol. Adjusted in the 16 FW to use the new freeze_time